### ISSUE TITLE
Question: Obtaining Modified Stats?

### ISSUE THREAD

**Issue by mmstick** [rust-lang/git2-rs#237]
I'm building [an Ion plugin](https://github.com/mmstick/ion-plugins/blob/master/src/namespaces/strings/git/lib.rs) to integrate stats from git repositories into the new namespaces plugin feature, but I'm having a hard time figuring out how to obtain the information I'm looking for from the documentation. Basically, how would one go about obtaining information like:

- Number of files that have been modified since the last commit
- Number of lines that have been added since the last commit
- Number of lines that have been removed since the last commit
- Obtaining the commit hash of the latest commit

Basically, Ion can currently call dynamic libraries and execute functions when given a variable like so:

```
echo ${namespace::function}
```

But I've only figured out, thus far, how to get the name of the currently-active branch.

```
echo ${git::branch}
```

**Comment by alexcrichton** [comment]
I think for diffs you'd use the various `Repository::diff_*` functions and then call `Diff::stats`. The latest commit and branch information can also be learned from the `Repository`

**Comment by alexcrichton** [comment]
I'm gonna close this since it's quite old, but if anything is missing a PR is certainly most welcome to add it!
